source/dataman/database/AMDatabase.cpp

```diff
@@ -39,24 +39,32 @@ QMutex AMDatabase::databaseLookupMutex_(QMutex::Recursive);
 
 // This constructor is protected. Only access is through AMDatabase::createDatabase().
  AMDatabase::~AMDatabase(){}
-AMDatabase::AMDatabase(const QString& connectionName, const QString& dbAccessString) :
-	QObject(),
+AMDatabase::AMDatabase(const QString& connectionName, const QString& dbAccessString, bool openAsReadOnly)
+	: QObject(),
 	connectionName_(connectionName),
 	dbAccessString_(dbAccessString),
 	qdbMutex_(QMutex::Recursive)
 {
-	QFileInfo accessInfo(dbAccessString_);
-	if(accessInfo.exists())
-		isReadOnly_ = !accessInfo.isWritable();
-	else
-		isReadOnly_ = false; // Probably not correct. If this the call that creates a new database, when does the .db file actually get created?
+
+	if (openAsReadOnly)
+		isReadOnly_ = true;
+
```

Comment:
This is a really minor quibble, but it seems odd to have the if() .. missing the braces but have the  else include them. Would be easy to screw up editing this section when maintaining this in future.